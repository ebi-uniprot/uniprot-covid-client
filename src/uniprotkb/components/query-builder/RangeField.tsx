import React, { Fragment } from 'react';
import { SearchTermType } from '../../types/searchTypes';

type RangeFieldProps = {
  field: SearchTermType;
  type?: string;
  handleChange: (value: string, isFrom: boolean) => void;
  rangeFrom?: string;
  rangeTo?: string;
};

const RangeField = ({
  field,
  type,
  handleChange,
  rangeFrom = '',
  rangeTo = '',
}: RangeFieldProps) => (
  <Fragment>
    <label htmlFor={`from_input_${field.id}`}>
      From
      <input
        id={`from_input_${field.id}`}
        data-testid="range-field-from-input"
        type={type}
        onChange={e => handleChange(e.target.value, true)}
        placeholder="0"
        value={rangeFrom}
      />
    </label>
    <label htmlFor={`to_input_${field.id}`}>
      To
      <input
        id={`to_input_${field.id}`}
        data-testid="range-field-to-input"
        type={type}
        onChange={e => handleChange(e.target.value, false)}
        placeholder="100"
        value={rangeTo}
      />
    </label>
  </Fragment>
);
export default RangeField;
