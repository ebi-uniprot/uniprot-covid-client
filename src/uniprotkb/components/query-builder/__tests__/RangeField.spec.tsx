import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RangeField from '../RangeField';

let props;
let rendered;

describe('Range field', () => {
  beforeEach(() => {
    props = {
      field: {
        id: 'range_field',
        field: {
          label: 'Any',
          itemType: 'feature',
          term: 'sites',
          dataType: 'string',
          hasRange: true,
          description: 'Search by feature sites',
          example: 'translocation',
        },
        queryInput: {},
      },
      handleChange: jest.fn(),
    };
    rendered = render(<RangeField {...props} />);
  });

  test('should render a range field', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should handle from/to change', () => {
    const { queryAllByTestId } = rendered;
    const fromNodes = queryAllByTestId('range-field-from-input');
    const toNodes = queryAllByTestId('range-field-to-input');
    const index = 0;
    fireEvent.change(fromNodes[index], { target: { value: 'val1' } });
    fireEvent.change(toNodes[index], { target: { value: 'val1' } });
    expect(props.handleChange).toHaveBeenCalledTimes(2);
  });
});
