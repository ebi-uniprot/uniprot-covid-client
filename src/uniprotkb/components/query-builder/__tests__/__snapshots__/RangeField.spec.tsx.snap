// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range field should render a range field 1`] = `
<DocumentFragment>
  <label
    for="from_input_range_field"
  >
    From
    <input
      data-testid="range-field-from-input"
      id="from_input_range_field"
      placeholder="0"
      value=""
    />
  </label>
  <label
    for="to_input_range_field"
  >
    To
    <input
      data-testid="range-field-to-input"
      id="to_input_range_field"
      placeholder="100"
      value=""
    />
  </label>
</DocumentFragment>
`;
